client/connect-account-page/index.tsx

```diff
@@ -52,26 +56,52 @@ const ConnectAccountPage: React.FC = () => {
 		connectUrl,
 		connect: { availableCountries, country },
 		devMode,
+		onboardingTestMode,
+		isJetpackConnected,
+		isAccountConnected,
+		isAccountValid,
 	} = wcpaySettings;
 
 	const isCountrySupported = !! availableCountries[ country ];
 
+	const urlParams = new URLSearchParams( window.location.search );
+
 	const determineTrackingSource = () => {
-		const urlParams = new URLSearchParams( window.location.search );
-		const from = urlParams.get( 'from' ) || '';
+		// If we have a source query param in the current request, use that.
+		const urlSource = urlParams.get( 'source' )?.replace( /[^\w-]+/g, '' );
+		if ( !! urlSource && 'unknown' !== urlSource ) {
```

Comment:
`!!` is this a mistake? I think it should be just `!`

Reply:
I want to test if the urlSource is truth and not `unknown`